src/Options.php

```diff
@@ -0,0 +1,307 @@
+                if (preg_match('/^'.$domain.'$/i', $value)) {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        return in_array($value, $this->lists[$listName][$type], true);
+    }
+
+    /**
+     * Returns a specific list.
+     *
+     * @param 'whitelist'|'blacklist' $listName
+     * @param string|null $type
+     *
+     * @throws InvalidOptionException
+     *
+     * @return array
+     */
+    public function getList($listName, $type = null)
+    {
+        $this->checkListByName($listName);
+
+        if (null !== $type) {
+            if (!array_key_exists($type, $this->lists[$listName])) {
+                throw new InvalidOptionException('Provided type "'.$type.'" must be "ip", "port", "domain" or "scheme"');
+            }
+
+            return $this->lists[$listName][$type];
+        }
+
+        return $this->lists[$listName];
+    }
+
+    /**
+     * Sets a list, the values must be passed as an array.
+     *
+     * @param 'whitelist'|'blacklist' $listName
+     * @param array $values
+     * @param string|null $type
+     *
+     * @throws InvalidOptionException
+     *
+     * @return Options
+     */
+    public function setList($listName, $values, $type = null)
+    {
+        $this->checkListByName($listName);
+
+        if (!is_array($values)) {
+            throw new InvalidOptionException('Provided values must be an array, "'.gettype($values).'" given');
+        }
+
+        if (null !== $type) {
+            if (!array_key_exists($type, $this->lists[$listName])) {
+                throw new InvalidOptionException('Provided type "'.$type.'" must be "ip", "port", "domain" or "scheme"');
+            }
+
+            $this->lists[$listName][$type] = $values;
+
+            return $this;
+        }
+
+        foreach ($values as $type => $value) {
+            if (!in_array($type, ['ip', 'port', 'domain', 'scheme'], true)) {
+                throw new InvalidOptionException('Provided type "'.$type.'" must be "ip", "port", "domain" or "scheme"');
+            }
+
+            $this->lists[$listName][$type] = $value;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Adds a value/values to a specific list.
+     *
+     * @param 'whitelist'|'blacklist' $listName
+     * @param string $type
+     * @param array|string $values
+     *
+     * @throws InvalidOptionException
+     *
+     * @return Options
+     */
+    public function addToList($listName, $type, $values)
+    {
+        $this->checkListByName($listName);
+
+        if (!array_key_exists($type, $this->lists[$listName])) {
+            throw new InvalidOptionException('Provided type "'.$type.'" must be "ip", "port", "domain" or "scheme"');
+        }
+
+        if (empty($values)) {
+            throw new InvalidOptionException('Provided values cannot be empty');
+        }
+
+        //Cast single values to an array
+        $values = (array) $values;
```

Comment:
Don't you need to check if it's an array first with something like that:

```php
if (!is_array($values)) {
    $values = [$values];
}
```

Reply:
for me, it is the same but I prefer this because it is less verbose

```
>>> $values = 'onevalue';
=> "onevalue"
>>> (array) $values;
=> [
     "onevalue",
   ]
>>> $values = ['onevalue'];
=> [
     "onevalue",
   ]
>>> (array) $values;
=> [
     "onevalue",
   ]
>>> 
```